add docs for shards' capacity health indicator (#94738)
elastic/elasticsearch: docs/reference/health/health.asciidoc

    The `invocations_since_last_success` key will report a map where the unhealthy policy
    name is the key and it's corresponding number of failed invocations is the value.

[[health-api-response-details-shards-usage]]
===== shards_capacity
`data`::

> (map) A view with information about the current capacity of shards for data nodes that do not belong to the frozen layer.

Reviewer: ```suggestion
(map) A view with information about the current capacity of shards for data nodes that do not belong to the frozen tier.
```

+
.Properties of `data`
[%collapsible%open]
=====
`max_shards_in_cluster`::
(int) Indicates the maximum number of shards that the cluster can hold.

`current_used_shards`::
(optional, int) The total number of shards hold by the cluster. Only displayed in the case the indicator's status is `red` or `yellow`.

=====

`frozen`::

> (map) A view with information about the current capacity of shards for data nodes that belong to the frozen layer.

Reviewer: ```suggestion
(map) A view with information about the current capacity of shards for data nodes that belong to the frozen tier.
```

+
.Properties of `frozen`
[%collapsible%open]
=====
`max_shards_in_cluster`::

> (int) Indicates the maximum number of shards that the cluster can hold.

Reviewer: Should we say these are the frozen shards?

> (int) Indicates the maximum number of shards that the cluster can hold.

Reviewer: ```suggestion
(int) Indicates the maximum number of shards the cluster can hold for the partially mounted indices.
```
